### PR TITLE
Improve cmaps

### LINKED ISSUE #488: Rework the colorMaps routines
Simplify the colormap picking routine.
Enable discrete and continuous cmaps
enable level picking in the ini file

### PR DESCRIPTION
**Pull request by matthiasto**
Refactor the makeColorMap function
fix #488

### CODE REVIEW

File: avaframe/out3Plot/plotUtils.py [hunk 1]
```diff
@@ -191,19 +182,119 @@
...
+def makeColorMap(colormapDict, levMin, levMax, continuous=False):
+    """Get colormap for plot
+
+    get the colormap, norm, levels... for ploting results
+
+    colormapDict is a dict as described in the Parameters section or a matplotlib
+    cmap. In the second case, the continuous flag is ignored and a continuous cmap is returned with
+    levelsNew and colorsNew set to None.
+    In the first case, depending on the continuous flag and what is given in the colormapDict
+    dictionary, the cmap is created and the levelsNew and colorsNew are created.
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:207]: suggestion: 
Options: 
1. colormapDict = dict (details see Parameters) - depending on the continuous flag and what is given in the colormapDict dictionary, the cmap is created and the levelsNew and colorsNew are created.
2. colormapDict = matplotlib cmap - the continuous flag is ignored and a continuous cmap is returned with
    levelsNew and colorsNew set to None.

File: avaframe/out3Plot/plotUtils.py [hunk 2]
```diff
@@ -191,19 +182,119 @@
...
+
+    Parameters
+    ----------
+        colormapDict: dict or cmap
+            cmap: matplotlib colormap
+                continuous colormap
+            colors: list
+                list of hex or rgb or dec colors
+            levels: list
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:216]: if colormap= dict , add this info before listing the dict keys

File: avaframe/out3Plot/plotUtils.py [hunk 3]
```diff
@@ -191,19 +182,119 @@
...
+    Returns
+    -------
+        cmap: matplotlib colormap
+            new color map
+        colorsNew: list
+            new list of hex or rgb or dec colors (None for continuous=True)
+        levelsNew: list
+            new levels corresponding to the colors (one more level than colors)
+        norm: matplotlib norm
+            norm associated to the levels and the colors (None for continuous=True)
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:234]: please update

File: avaframe/out3Plot/plotUtils.py [hunk 4]
```diff
@@ -191,19 +182,119 @@
...
+    """
+
+    if type(colormapDict) is matplotlib.colors.LinearSegmentedColormap:
+        cmap = colormapDict
+        colorsNew = None
+        norm = None
+        levelsNew = None
+    elif continuous:
+        # do we have a cmap in the dict?
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:243]: suggestion: check if cmap is provided

File: avaframe/out3Plot/plotUtils.py [hunk 5]
```diff
@@ -191,19 +182,119 @@
...
+        if 'cmap' in colormapDict.keys():
+            cmap = colormapDict['cmap']
+            colorsNew = None
+        # if not do we have a list of colors in the dict?
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:247]: check if list of colors is provided

File: avaframe/out3Plot/plotUtils.py [hunk 6]
```diff
@@ -191,19 +182,119 @@
...
+            raise FileNotFoundError(message)
+
+        try:
+            indEnd = np.where(np.asarray(levels) >= levMax)[0][0]
+        except IndexError:
+            indEnd = len(levels)
+        levelsNew = levels[:indEnd]
+        levelsNew.append(levMax)
+
+        # do we have a list of colors in the dict?
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:274]: same here

File: avaframe/out3Plot/plotUtils.py [hunk 7]
```diff
@@ -191,19 +182,119 @@
...
+        if 'colors' in colormapDict.keys():
+            colors = colormapDict['colors']
+            colorsNew = colors[:indEnd]
+            colorsNew.append(colors[indEnd-1])
+            cmap = mplCol.ListedColormap(colorsNew)
+            cmap.set_bad(color='w')
+            cmap.set_under(color='w')
+            # cmap.set_over(color='w')
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:282]: still needed?

File: avaframe/out3Plot/plotUtils.py [hunk 8]
```diff
@@ -191,19 +182,119 @@
...
+        # if not do we have a cmap in the dict?
```
Comment (awirb) [inline avaframe/out3Plot/plotUtils.py:283]: check if a cmap is provided

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/out3Plot/plotUtils.py [hunk 9]
```diff
@@ -100,38 +100,35 @@
 cmapViridis.set_bad(color='k')
 
 # divergent color map
-cmapdiv = copy.copy(matplotlib.cm.RdBu_r)  # sns.color_palette("RdBu_r")
-
+# cmapdiv = cmapCameri.vik
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:103]: Please remove.

**Review by fso42 — APPROVED** [review]

**Pull request merged**